External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -2193,12 +2184,13 @@ void OpDispatchBuilder::BSFOp(OpcodeArgs) {
       Src, ZeroConst,
       OneConst, ZeroConst);
 
-  StoreResult(GPRClass, Op, SelectOp, -1);
+  StoreResult_WithOpSize(GPRClass, Op, Op->Dest, SelectOp, DstSize, -1);
   SetRFLAG<FEXCore::X86State::RFLAG_ZF_LOC>(ZFSelectOp);
 }
 
 void OpDispatchBuilder::BSROp(OpcodeArgs) {
-  OrderedNode *Dest = LoadSource(GPRClass, Op, Op->Dest, Op->Flags, -1);
```

Comment:
what about size==4 here?

Reply:
That's actually explicitly the operation that this change is working around.
When size == 4, we need to be careful to not do a partial writeback of the register when the incoming source has zero bits set, so we load the full 8byte register in that case to ensure zext magic doesn't occur in StoreResult

Reply:
Cool beans